columnflow/tasks/histograms.py

```diff
@@ -221,18 +228,29 @@ def run(self):
                         # enable weights and store it
                         histograms[var_key] = h.Weight()
 
+                    # create event mask
+                    masks = []
+                    for variable_inst in variable_insts:
+                        sel = variable_inst.selection
+                        if sel != "1":
+                            if not callable(sel):
+                                raise ValueError(f"invalid selection '{sel}', for now only callables are supported")
+                            masks.append(sel(events))
+                    mask = reduce(and_, masks, np.ones(len(events), dtype=bool))
+                    masked_events = events[mask]
```

Comment:
Could we reduce events here iteratively?

```suggestion
                    # mask events and weights when selection expressions are found
                    masked_events = events
                    masked_weights = weights
                    for variable_inst in variable_insts:
                        sel = variable_inst.selection
                        if sel == "1":
                            continue
                        if not callable(sel):
                            raise ValueError(f"invalid selection '{sel}', for now only callables are supported")
                        mask = sel(masked_events)
                        masked_events = masked_events[mask]
                        masked_weights = masked_weights[mask]
```

Not that this has an overly large positive impact on performance (although a small one for sure), but it would be good to consistently start using this pattern :)

If we decide for that, the imports above and the use of `weights` below needed to be adapted as well.